core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -315,6 +315,36 @@ class DAGScheduler(
     parents.toList
   }
 
+  /**
+   * Get partitions of parent rdds for a given RDD.
+   * Traverses the lineage chain and calls partitions on each RDD.
+   */
+  private[scheduler] val shuffleToRDDParts = new HashSet[Int]
+  private def getParentPartitions(rdd: RDD[_]) = {
+    val visited = new HashSet[RDD[_]]
+    val waitingForVisit = new Stack[RDD[_]]
+    def visit(r: RDD[_]) {
+      if (!visited(r)) {
+       visited += r
+       for (dep <- r.dependencies) {
+         dep match {
+           case shuffleDep: ShuffleDependency[_, _, _] =>
+             if (!shuffleToRDDParts(shuffleDep.shuffleId)) {
+               shuffleDep.rdd.partitions
```

Comment:
It's not obvious why this is being called here.  I think it would be cleaner to rename this method something like "getParentRDDs that returns a `List[RDD]`, then map over the result to call `partitions`.